cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -733,7 +733,10 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
                         if (Annotations.booleanAnnotation(sts, Annotations.ANNO_STRIMZI_IO_MANUAL_ROLLING_UPDATE,
                                 false, Annotations.ANNO_OP_STRIMZI_IO_MANUAL_ROLLING_UPDATE)) {
                             return kafkaSetOperations.maybeRollingUpdate(sts, pod -> {
-
+                                if (pod == null) {
+                                    log.debug("Pod not found, no need to roll");
+                                    return "";
```

Comment:
It would be good to understand how we can get into this situation. My concern is that maybe it's fine not to roll the pod if it (momentarily) doesn't exist. But does that mean we don't wait for its readiness (I think we would wait for it's readiness, but will an NPE pop up in that code)?

Reply:
Looking at the affected ST I think we are waiting for the pod readiness correctly.
This issue appears if these two STs are run in this order: `io.strimzi.systemtest.RollingUpdateST#testKafkaWontRollUpBecauseTopic, io.strimzi.systemtest.RollingUpdateST#testExternalLoggingChangeTriggerRollingUpdate`

Reply:
That's an observation, not an explanation. What is it about running those tests in that order which causes the failure? Understanding the real reason is important. It could be an issue with the tests. It could be we find out about a much more general problem with the operator. If you never understand the true reason we'll never know.

Reply:
I don't think it's problem in the tests since we only delete Kafka CR and waiting for deletion. We do force deletion of pods every 10 seconds (we had issues with an automatic deletion on some environments), it might be caused by it even when CR is no longer available?

Reply:
AFAIK these are just race conditions when the Pods are deleted by garbage collection while reconciliation is running?